<launch>

  <include file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="rgb_frame_id" value="head_camera_rgb_optical_frame" />
    <arg name="depth_frame_id" value="head_camera_depth_optical_frame" />
    <arg name="depth_registration" value="true" />
    <arg name="publish_tf" value="false" />
    <arg name="camera" value="head_camera" />
  </include>

  <group ns="head_camera">
    <!-- decimated to 160x120 -->
    <node pkg="nodelet" type="nodelet" name="crop_decimate" args="load image_proc/crop_decimate /head_camera/head_camera_nodelet_manager" output="screen">
      <remap from="camera/image_raw" to="depth_registered/image_raw" />
      <remap from="camera/camera_info" to="depth_registered/camera_info" />
      <remap from="camera_out" to="depth_downsample" />
      <param name="decimation_x" value="4" />
      <param name="decimation_y" value="4" />
      <param name="queue_size" value="1" />
    </node>
    <!-- downsampled XYZ point cloud (mainly for navigation) -->
    <!-- ARD: somewhere there's a ValueError: namespace must be global -->
    <node pkg="nodelet" type="nodelet" name="points_downsample"
          args="load depth_image_proc/point_cloud_xyz /head_camera/head_camera_nodelet_manager"
          ns="depth_downsample">
      <remap from="image_rect" to="image_raw"/>
    </node>
  </group>

</launch>
